mantidproject/mantid
# ALC Loading interface to MVP

### Description of work
Changed code for ALC Loading interface to follow the MVP architecture.

#### Summary of work

- Created abstract class for presenter
- Made presenter NOT a QObject
- Put all Qt code inside the view
- Created new class for model
- Created abstract class for model
- Made abstract class of view NOT a QObject
- Corrected presenter unit test



Followed the requests stated in #38009 .

Fixes #38009 . 




#### Further detail of work


### To test:
- Code review
- Manual testing (Need to have directory with training dataset):
  - Open Muon > ALC interface
  - Change instrument to MUSR
  - For the runs, type 15189
  - You should see some green text indicating the run has been found
  - Click in manage user directories, check that interface opens
  - Click in Period info, check that it opens
  - Click Load
  - After loading, click Auto add
  - Go to the local directory where the file is being loaded from and copy the `MUSR15189` file and rename the copy to `MUSR15190`
  - You should see the interface loads the file and the runs text box is now updated with the run 15189
  - Go to the data modelling page
  - You should see the same data displayed as in the data loading interface







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
